diff --git a/src/arch/x86/SConscript b/src/arch/x86/SConscript
index 4fabe434f5..a1901d310c 100644
--- a/src/arch/x86/SConscript
+++ b/src/arch/x86/SConscript
@@ -53,7 +53,9 @@ Source('emulenv.cc', tags=['x86 isa'])
 Source('faults.cc', tags=['x86 isa'])
 Source('fs_workload.cc', tags=['x86 isa'])
 Source('insts/badmicroop.cc', tags=['x86 isa'])
+Source('htm.cc', tags=['x86 isa'])
+Source('insts/htm.cc', tags=['x86 isa'])
 Source('insts/microop.cc', tags=['x86 isa'])
 Source('insts/microregop.cc', tags=['x86 isa'])
 Source('insts/static_inst.cc', tags=['x86 isa'])
