diff --git a/src/arch/x86/isa/includes.isa b/src/arch/x86/isa/includes.isa
index 6cc3674998..e6919bdba7 100644
--- a/src/arch/x86/isa/includes.isa
+++ b/src/arch/x86/isa/includes.isa
@@ -65,6 +65,7 @@
 #include "arch/x86/insts/microspecop.hh"
 #include "arch/x86/insts/static_inst.hh"
 #include "arch/x86/isa.hh"
+#include "arch/x86/insts/htm.hh"
 #include "arch/x86/regs/ccr.hh"
 #include "arch/x86/regs/int.hh"
 #include "arch/x86/regs/misc.hh"
