diff --git a/backends/tm_impl/tiny_stm/TinySTM_runtime.cpp b/backends/tm_impl/tiny_stm/TinySTM_runtime.cpp
--- a/backends/tm_impl/tiny_stm/TinySTM_runtime.cpp
+++ b/backends/tm_impl/tiny_stm/TinySTM_runtime.cpp
@@ -125,6 +125,12 @@ static void do_tm_exit()
 	fprintf(stdout, " aborts=%llu\n", (unsigned long long)ac);
 	fflush(stdout);
 
+#ifndef NDEBUG
+	fprintf(stderr,
+	        "\n=== TinySTM max read-set = %llu, max write-set = %llu ===\n",
+	        (unsigned long long)mxr,
+	        (unsigned long long)mxw);
+#endif
 	if (ac > 0) {
 		fprintf(stderr,
 		        "\n=== TinySTM total aborts = %llu ===\n",
@@ -193,6 +199,12 @@ void tm_exit()
 	fprintf(stdout, " aborts=%llu\n", (unsigned long long)ac);
 	fflush(stdout);
 
+#ifndef NDEBUG
+	fprintf(stderr,
+	        "\n=== TinySTM max read-set = %llu, max write-set = %llu ===\n",
+	        (unsigned long long)mxr,
+	        (unsigned long long)mxw);
+#endif
 	if (ac > 0) {
 		fprintf(stderr,
 		        "\n=== TinySTM total aborts = %llu ===\n",
